app-profile-update (#986)
spectrocloud/librarium: content/docs/04.5-devx/05-app-profile.md

## App Profiles

> App Profiles are templates created with preconfigured services required for sandbox cluster deployment. App Profiles provide a way to drive consistency across sandbox clusters. You can create as many profiles as required. An App Profile can be created to meet specific types of Workload on Palette Sandbox Clusters. For example, you may create a Development App Profile with <Tooltip trigger={<u>Helm Charts</u>}>Add a Palette App profile Pack from Helm registry charts. The public Palette Pack registry and a few popular Helm chart repositories are already available out of the box. Add additional pack registries or public/private chart registries to Palette. </Tooltip>

Reviewer: ```suggestion
App Profiles are templates created with preconfigured services required for sandbox cluster deployment. App Profiles provide a way to drive consistency across sandbox clusters. You can create as many profiles as required. An App Profile can be created to meet specific types of Workload on Palette Virtual Clusters. For example, you may create a Development App Profile with <Tooltip trigger={<u>Helm Charts</u>}>Add a Palette App profile Pack from Helm registry charts. The public Palette Pack registry and a few popular Helm chart repositories are already available out of the box. Add additional pack registries or public/private chart registries to Palette. </Tooltip>
```

> , <Tooltip trigger={<u>Manifest</u>}>Layers can be constructed using raw manifests to provision Kubernetes resources unavailable via Palette or Charts. Pack Manifests provide a pass-through mechanism wherein additional Kubernetes resources can be orchestrated onto a cluster and the rest of the stack. Specific integrations may require the creation of Secrets or CustomResourceDefinition (CRDs); manifest files can be attached to the layer to achieve this. </Tooltip>, different database applications, etc. 

Reviewer: ```suggestion
, <Tooltip trigger={<u>Manifest</u>}>Layers can be constructed using raw manifests to provision Kubernetes resources that are unavailable in Palette or Charts. Pack Manifests provide a pass-through mechanism in which additional Kubernetes resources can be orchestrated onto a cluster and the rest of the stack. Specific integrations may require the creation of Secrets or CustomResourceDefinition (CRDs). Manifest files can be attached to the layer to achieve this. For example, you could attach a Manifest to different database applications.
```

|Description (optional)   | Description of the App Profile, if any | 
|Tag (optional)               | Tags on a cluster group are propagated to the cloud/datacenter environments.|

3. **Add Service**: Select a service to start configuring the App Profile from the available services and configure the service based on your requirements. The following services are provided:

>   * Container Deployment: Containers are methods of building, packaging, and deploying an application. A container includes the code, runtime, libraries, and all the dependencies required by a containerized workload. Container deployment deploys containers to their target environment.

Reviewer 2: @resmaa - this description looks good for containers. Will we have a seperate page or section which talks about attributes of the Container deployment?

You can talk to @shubhamrajvanshi incase you need more info about each supported attribute.

|Description (optional)   | Description of the App Profile, if any | 
|Tag (optional)               | Tags on a cluster group are propagated to the cloud/datacenter environments.|

> 3. **Add Service**: Select a service to start the configuration of the App Profile from the available services and configure the service per the use case. The users can go for the following key services:

Reviewer: ```suggestion
3. **Add Service**: Select a service to start configuring the App Profile from the available services and configure the service based on your requirements. The following services are provided:
```

>   * Container Deployment: Containers are methods of building, packaging, and deploying an application. A container includes the code, runtime, libraries, and all the dependencies required by a containerized workload. Container deployment deploys containers to their target environment.

Reviewer: ```suggestion
  * Container Deployment: Containers are methods of building, packaging, and deploying an application. A container includes the code, runtime libraries, and all the dependencies required by a containerized workload. Container deployment deploys containers to their target environment.
```

 

 * [Helm](/devx/registries#palettehelmregistry): Palette App profile Pack from Helm registry charts. The public Palette Pack registry and a few popular Helm chart repositories are already available out of the box. Add additional pack registries or public or private chart registries to Palette.
 

>  * Manifest: Layers can be constructed using raw manifests to provision Kubernetes resources unavailable via Palette or Charts. Pack Manifests provide a pass-through mechanism wherein additional Kubernetes resources can be orchestrated onto a cluster and the rest of the stack. Specific integrations may require the creation of Secrets or CustomResourceDefinition (CRDs); manifest files can be attached to the layer to achieve this.

Reviewer: ```suggestion
 * Manifest: Layers can be constructed using raw manifests to provision Kubernetes resources that are unavailable in Palette or Charts. Pack Manifests provide a pass-through mechanism in which additional Kubernetes resources can be orchestrated onto a cluster and the rest of the stack. Specific integrations may require the creation of Secrets or CustomResourceDefinition (CRDs). Manifest files can be attached to the layer to achieve this. 
```

>  *  Database Services: Different database applications such as MongoDB, PostgreSQl, Redis, etc.

Reviewer: Between steps 3 and 4 we agreed to add a Warning box about namespace. Below I offered a couple variations of text for you to consider because you're more familiar with the product. Also, should "namespace" be bold?

_You suggested:_ "When adding layers to the App profile using manifest files, specify a **namespace** name to which the resource must belong."

_Another suggestion with slight change:_ "When adding layers to the App profile using manifest files, you must specify a **namespace** name to which the resource belongs."

_Another suggestion:_ "When adding layers to the App profile using manifest files, you must specify a **namespace** name to indicate which resource the layer belongs to."
